sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/identity/impl/utils/ClassUtil.java

```diff
@@ -0,0 +1,202 @@
+                baseClass, Arrays.stream(constructorArguments).map(Object::getClass).map(u -> getQualifiedName(u))
+                    .collect(Collectors.joining(", ")))));
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <T> Class<T> getClass(String className, Class<TokenCredentialProvider> assignableClass) {
+        if (className != null && !className.isEmpty()) {
+            try {
+                Class<?> clazz = Class.forName(className);
+                if (!assignableClass.isAssignableFrom(clazz)) {
+                    throw new AzureException("Provided class [" + className + "] is not a [ " + assignableClass.getSimpleName() + "]");
+                }
+                return (Class<T>) clazz;
+            } catch (ClassNotFoundException e) {
+                throw new AzureException("The provided class [" + className + "] can't be found", e);
+            }
+        }
+        return null;
+    }
+
+    private static Optional<Constructor<?>> findConstructor(Class<?> type, Object... constructorArguments) {
+
+        notNull(type, "Target type must not be null");
+        notNull(constructorArguments, "Constructor arguments must not be null");
+
+        return Arrays.stream(type.getDeclaredConstructors())//
+            .filter(constructor -> argumentsMatch(constructor.getParameterTypes(), constructorArguments))//
+            .findFirst();
+    }
+
+    private static boolean argumentsMatch(Class<?>[] parameterTypes, Object[] arguments) {
+
+        if (parameterTypes.length != arguments.length) {
+            return false;
+        }
+
+        int index = 0;
+
+        for (Class<?> argumentType : parameterTypes) {
+
+            Object argument = arguments[index];
+
+            // Reject nulls for primitives
+            if (argumentType.isPrimitive() && argument == null) {
+                return false;
+            }
+
+            // Type check if argument is not null
+            if (argument != null && !isAssignableValue(argumentType, argument)) {
+                return false;
+            }
+
+            index++;
+        }
+
+        return true;
+    }
+
+    private static  <T> T instantiateClass(Constructor<T> ctor, Object... args)  {
+        notNull(ctor, "Constructor must not be null");
+        try {
+            makeAccessible(ctor);
+            Class<?>[] parameterTypes = ctor.getParameterTypes();
+            isTrue(args.length <= parameterTypes.length, "Can't specify more arguments than constructor parameters");
+            Object[] argsWithDefaultValues = new Object[args.length];
+            for (int i = 0 ; i < args.length; i++) {
+                if (args[i] == null) {
+                    Class<?> parameterType = parameterTypes[i];
+                    argsWithDefaultValues[i] = (parameterType.isPrimitive() ? DEFAULT_TYPE_VALUES.get(parameterType) : null);
+                }
+                else {
```

Comment:
There are two styles of the curly bracket in this class, maybe we should stick to one style?
```java
if () {
} else {
}
```

Reply:
ok